src/Microsoft.AspNet.Mvc.Core/SessionStateTempDataProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Interfaces;
+using Microsoft.Framework.Internal;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Provides session-state data to the current <see cref="ITempDataDictionary"/> object.
+    /// </summary>
+    public class SessionStateTempDataProvider : ITempDataProvider
+    {
+        private static string TempDataSessionStateKey = "__ControllerTempData";
+
+        /// <inheritdoc />
+        public virtual IDictionary<string, object> LoadTempData([NotNull] HttpContext context)
+        {
+            if (!IsSessionEnabled(context))
+            {
+                // Session middleware is not enabled. No-op
+                return null;
+            }
+
+            var session = context.Session;
+            byte[] value = null;
```

Comment:
BTW don't need to set this to `null` because in all code paths it will be definitely assigned.